src/org/zaproxy/zap/extension/ascanrulesAlpha/SQLInjectionMsSQL.java

```diff
@@ -0,0 +1,319 @@
+		ORIG_VALUE_TOKEN + ")) ' WAITFOR DELAY '" + SLEEP_TOKEN + "'" + SQL_ONE_LINE_COMMENT,
+		ORIG_VALUE_TOKEN + ")) \" WAITFOR DELAY '" + SLEEP_TOKEN + "'" + SQL_ONE_LINE_COMMENT,
+		ORIG_VALUE_TOKEN + ") WAITFOR DELAY '" + SLEEP_TOKEN + "' (" ,
+		ORIG_VALUE_TOKEN + ") ' WAITFOR DELAY '" + SLEEP_TOKEN + "' (",
+		ORIG_VALUE_TOKEN + ") \" WAITFOR DELAY '" + SLEEP_TOKEN + "' (",
+		ORIG_VALUE_TOKEN + ")) WAITFOR DELAY '" + SLEEP_TOKEN + "' ((",
+		ORIG_VALUE_TOKEN + ")) ' WAITFOR DELAY '" + SLEEP_TOKEN + "' ((",
+		ORIG_VALUE_TOKEN + ")) \" WAITFOR DELAY '" + SLEEP_TOKEN + "' ((",
+	};
+
+	/**
+	 * plugin dependencies (none! not even "SQL Injection")
+	 */
+	private static final String[] dependency = {};
+
+	/**
+	 * for logging.
+	 */
+	private static Logger log = Logger.getLogger(SQLInjectionMsSQL.class);
+
+	/**
+	 * determines if we should output Debug level logging
+	 */
+	private boolean debugEnabled = log.isDebugEnabled();
+
+	@Override
+	public int getId() {
+		return 40027;
+	}
+
+	@Override
+	public String getName() {
+		return Constant.messages.getString("ascanalpha.sqlinjection.mssql.name");
+	}
+
+	@Override
+	public String[] getDependency() {
+		return dependency;
+	}
+
+	@Override
+	public boolean targets(TechSet technologies) {
+		return technologies.includes(Tech.MsSQL);
+	}
+
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString("ascanalpha.sqlinjection.mssql.desc");
+	}
+
+	@Override
+	public int getCategory() {
+		return Category.INJECTION;
+	}
+
+	@Override
+	public String getSolution() {
+		return Constant.messages.getString("ascanalpha.sqlinjection.mssql.soln");
+	}
+
+	@Override
+	public String getReference() {
+		return Constant.messages.getString("ascanalpha.sqlinjection.mssql.refs");
+	}
+
+	@Override
+	public void init() {
+		if ( this.debugEnabled ) log.debug("Initialising");
+
+		//set up what we are allowed to do, depending on the attack strength that was set.
+		if ( this.getAttackStrength() == AttackStrength.LOW ) {
+			doTimeBased=true; doTimeMaxRequests=3;
+		} else if ( this.getAttackStrength() == AttackStrength.MEDIUM) {
+			doTimeBased=true; doTimeMaxRequests=6;
+		} else if ( this.getAttackStrength() == AttackStrength.HIGH) {
+			doTimeBased=true; doTimeMaxRequests=12;
+		} else if ( this.getAttackStrength() == AttackStrength.INSANE) {
+			doTimeBased=true; doTimeMaxRequests=100;
+		}
+
+		// Read the sleep value from the configs
+		try {
+			this.sleep = this.getConfig().getInt("rules.common.sleep", 5);
+		} catch (ConversionException e) {
+			log.debug("Invalid value for 'rules.common.sleep': " + this.getConfig().getString("rules.common.sleep"));
+		}
+		if ( this.debugEnabled ) {
+			log.debug("Sleep set to " + sleep + " seconds");
+		}
+	}
+
+	/**
+	 * scans for SQL Injection vulnerabilities, using MsSQL specific syntax.  If it doesn't use specifically MsSQL syntax, it does not belong in here, but in TestSQLInjection
```

Comment:
`If it doesn't use specifically MsSQL syntax...` could be removed (it's already mentioned in the class' JavaDoc).